### ISSUE TITLE
Broken Check Cache Stage with Minikube and Buildkit

### ISSUE THREAD

**Issue by euriostigue** [GoogleContainerTools/skaffold#5183]
### Expected behavior
```bash
$ skaffold build
Generating tags...
 - wuher -> wuher:latest
 - raze -> raze:latest
 - sync-kinesis -> sync-kinesis:latest
Checking cache...
 - wuher: Not found. Building
 - raze: Not found. Building
 - sync-kinesis: Not found. Building
Found [minikube] context, using local docker daemon.
Building [sync-kinesis]...
```

### Actual behavior
```bash
$ skaffold build
Generating tags...
 - wuher -> wuher:latest
 - raze -> raze:latest
 - sync-kinesis -> sync-kinesis:latest
Checking cache...
 - wuher: Error checking cache.
getting hash for artifact "wuher": getting dependencies for "wuher": parsing dockerfile "/Users/eric/effx/effunctions/Dockerfile": Dockerfile parse error line 39: Unknown flag: mount
```

The Dockerfile:
```
# syntax=docker/dockerfile:experimental
# ubuntu:latest as of 11/30
FROM ubuntu@sha256:fff16eea1a8ae92867721d90c59a75652ea66d29c05294e6e2f898704bdb8cf1 AS base

...

COPY go.mod go.sum ./
RUN --mount=type=ssh go mod download && go mod verify # Line 37
...
RUN --mount=type=cache,target=/root/.cache/go-build make build

CMD ["npx", "sls", "offline","--stage", "dev"]
```

### Information

- minikube v1.16.0
- Skaffold version: v1.17.2
- Operating system: 
```bash
$ sw_vers
ProductName:    macOS
ProductVersion: 11.1
BuildVersion:   20C69
```
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2beta9
kind: Config
metadata:
  name: effunctions
build:
  local:
    useBuildkit: true
  tagPolicy:
    sha256: {}
  artifacts:
    - image: wuher
      custom:
        buildCommand: ./build.sh wuher
        dependencies:
          dockerfile:
            path: Dockerfile
            buildArgs:
              APP: wuher
    - image: raze
      custom:
        buildCommand: ./build.sh raze
        dependencies:
          dockerfile:
            path: Dockerfile
            buildArgs:
              APP: raze
    - image: sync-kinesis
      context: offline/sync-kinesis
deploy:
  kubectl:
    manifests:
      - "deploy/manifests/*.yaml"
    defaultNamespace: effunctions
```

### Steps to reproduce the behavior

After upgrading from skaffold v1.16 to v1.17.2, the hashing for cached images is broken. The same combination of Dockerfile, skaffold.yaml and build.sh works just fine with skaffold v1.16.

I'm guessing that the `useBuildkit` flag isn't getting passed properly during the "check cache" stage. If I removed all `mount` flags, the image will still build using buildkit.

**Comment by zamai** [comment]
Same thing is happening to me.
extract of my Dockerfile: 
```
# syntax = docker/dockerfile:experimental

FROM golang:1.13 as builder
... 
RUN --mount=type=cache,target=/root/.cache/go-build \
     CGO_ENABLED=0 GOOS=linux go build  -mod=vendor -v -tags static_all -ldflags '-w -s' -o /api ./service_dir
```
fails with: 
```
failed to build: getting hash for artifact "_image_tag_": getting dependencies for "_image_tag_": parsing dockerfile "/path_to_dockerfile/Dockerfile": Dockerfile parse error line 21: Unknown flag: mount
```

docker version: 
```
Client: Docker Engine - Community
 Cloud integration: 1.0.4
 Version:           20.10.0
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        7287ab3
 Built:             Tue Dec  8 18:55:43 2020
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.0
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       eeddea2
  Built:            Tue Dec  8 18:58:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

Skaffold 1.17.2

Downgrading to Skaffold 1.16.0 resolved the issue.

**Comment by dimkr** [comment]
Same issue, same symptoms, Ubuntu 18.04 with `docker-ce` and any skaffold 1.17.x. 1.16.0 works.

**Comment by dimkr** [comment]
Probably fixed by #5052 and a duplicate of #5178 among others.

**Comment by euriostigue** [comment]
My issue was resolved by using the bleeding edge release.

I'm going to close this issue because no action needs to be taken.
